website/src/utils/groupFieldsByHeader.ts

```diff
@@ -0,0 +1,37 @@
+import type { InputField, Metadata } from '../types/config';
+
+const SUBMISSION_ID_FIELD: InputField = {
+    name: 'submissionId',
+    displayName: 'Submission ID',
+    definition: 'FASTA ID',
+    guidance:
+        'Your sequence identifier; should match the FASTA file header - this is used to link the metadata to the FASTA sequence',
+    example: 'GJP123',
+    noEdit: true,
+    required: true,
+};
+
+export const groupFieldsByHeader = (inputFields: InputField[], metadata: Metadata[]): Map<string, InputField[]> => {
```

Comment:
This grouping by Header will come in very handy also for the dropdowns in the Column remapping!

I'm wondering, is there a meaningful way we can sort the `InputField[]`s here? The way it comes out of this function now is it's just ordered the same way as in the config; is that good? For larger lists of fields, maybe alphabetical sorting is easier?

Not sure if you made any considerations regarding sorting yet (or maybe this is also not the right place to sort things), feel free to leave it as is!

Reply:
I think for the moment using the input order might be easiest as I can see us wanting to have more important fields first and they wouldn't necessarily be first alphabetically. I don't see a way to customize the order without adding a new metadata flag though (maybe you have a suggestion) - I'm not sure the order is important enough to complicate the values.yaml with yet another flag - but we can come back to it later if it becomes an issue :-)